components/style/properties/gecko.mako.rs

```diff
@@ -663,6 +661,116 @@ fn static_assert() {
                                corner.y_index,
                                need_clone=True) %>
     % endfor
+
+    pub fn set_border_image_source(&mut self, v: longhands::border_image_source::computed_value::T) {
+        unsafe {
+            // Prevent leaking of the last elements we did set
+            Gecko_SetNullImageValue(&mut self.gecko.mBorderImageSource);
+        }
+
+        if let Some(image) = v.0 {
+            // TODO: We need to make border-image-source match with background-image
+            // until then we are setting with_url to false
+            self.gecko.mBorderImageSource.set(image, false, &mut false)
+        }
+    }
+
+    pub fn copy_border_image_source_from(&mut self, other: &Self) {
+        unsafe {
+            Gecko_CopyImageValueFrom(&mut self.gecko.mBorderImageSource,
+                                     &other.gecko.mBorderImageSource);
+        }
+    }
+
+    pub fn set_border_image_outset(&mut self, v: longhands::border_image_outset::computed_value::T) {
+        % for side in SIDES:
+            v.${side.index}.to_gecko_style_coord(&mut self.gecko.mBorderImageOutset
+                                                          .data_at_mut(${side.index}));
+        % endfor
+    }
+
+    pub fn copy_border_image_outset_from(&mut self, other: &Self) {
+        % for side in SIDES:
+            self.gecko.mBorderImageOutset.data_at_mut(${side.index})
+                .copy_from(&other.gecko.mBorderImageOutset.data_at(${side.index}));
+        % endfor
+    }
+
+    pub fn set_border_image_repeat(&mut self, v: longhands::border_image_repeat::computed_value::T) {
+        use properties::longhands::border_image_repeat::computed_value::RepeatKeyword;
+        use gecko_bindings::structs::{NS_STYLE_BORDER_IMAGE_REPEAT_STRETCH, NS_STYLE_BORDER_IMAGE_REPEAT_REPEAT};
+        use gecko_bindings::structs::{NS_STYLE_BORDER_IMAGE_REPEAT_ROUND, NS_STYLE_BORDER_IMAGE_REPEAT_SPACE};
+        let mut keywords = vec![];
+
+            % for side in range(2):
```

Comment:
Don't create a vector here, just set directly. `for i,side in enumerate(["H", "V"])` will help.